AmplifyPlugins/API/AWSAPICategoryPlugin/Configuration/AWSAPICategoryPluginConfiguration+EndpointConfig.swift

```diff
@@ -281,31 +281,26 @@ public extension AWSAPICategoryPluginConfiguration {
 }
 
 extension Dictionary where Key == String, Value == AWSAPICategoryPluginConfiguration.EndpointConfig {
-    func getConfig(for apiName: String?,
-                   endpointType: AWSAPICategoryPluginEndpointType) throws ->
+
+    /// Getting the `EndpointConfig` resolves to the following rules:
+    /// 1. If `apiName` is specified, retrieve the endpoint configuration for this api
+    /// 2. If `apiName` is not specified, and `endpointType` is, retrieve the endpoint if there is only one.
+    /// 3. If nothing is specified, return the endpoint only if there is a single one, with GraphQL taking precedent
+    /// over REST.
+    func getConfig(for apiName: String? = nil,
+                   endpointType: AWSAPICategoryPluginEndpointType? = nil) throws ->
         AWSAPICategoryPluginConfiguration.EndpointConfig {
         if let apiName = apiName {
             return try getConfig(for: apiName)
         }
-
-        let apiForEndpointType = filter { (_, endpointConfig) -> Bool in
-            return endpointConfig.endpointType == endpointType
-        }
-
-        guard let endpointConfig = apiForEndpointType.first else {
-            throw APIError.invalidConfiguration("Missing API for \(endpointType) endpointType",
-                                                "Add the \(endpointType) API to configuration.")
-        }
-
-        if apiForEndpointType.count > 1 {
-            throw APIError.invalidConfiguration(
-                "More than one \(endpointType) API configured. Could not infer which API to call",
-                "Use the apiName to specify which API to call")
+        if let endpointType = endpointType {
+            return try getConfig(for: endpointType)
         }
 
-        return endpointConfig.value
+        return try getConfig()
     }
 
+    // Retrieve the endpoint configuration for the specified `apiName`
```

Comment:
nit: doc comment
```suggestion
    /// Retrieve the endpoint configuration for the specified `apiName`
```

But do we need it? The method is pretty self-documenting

Reply:
yes for this one i'll remove, the other ones below do provide a bit more context, such as taking GraphQL over REST